### PR TITLE
Added support for http intercept of device code request

### PR DESCRIPTION
**Pull request by fadidurah**
Making small PR to add support for intercepting the device code http request made in device code flow in test cases. Related to bug https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1509622/

### CODE REVIEW

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/mocks/MockTokenCreator.java [hunk 1]
```diff
@@ -64,6 +64,7 @@
     public static final String MOCK_ISSUER_SUFFIX_VALUE = "/v2.0";
     public static final Pattern CLOUD_DISCOVERY_ENDPOINT_REGEX = Pattern.compile("^https:\\/\\/login.microsoftonline.com\\/common\\/discovery\\/instance\\?api-version=1.1\\&authorization_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize$");
     public static final Pattern MOCK_TOKEN_URL_REGEX = Pattern.compile("https:\\/\\/login.microsoftonline.com\\/.*");
+    public static final Pattern DEVICE_CODE_FLOW_AUTHORIZATION_REGEX = Pattern.compile("https:\\/\\/login.microsoftonline.com\\/common\\/oAuth2\\/v2.0\\/devicecode");
```
Comment (AdamBJohnsonx) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/mocks/MockTokenCreator.java:67]: Feels like I should see a reference to this constant somewhere...

Comment (fadidurah) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/mocks/MockTokenCreator.java:67]: It's in SingleAccountOverloadsMockedTest.java, a testing class in msal for single account

**Comment by AdamBJohnsonx** [comment]
You probably want to update your common submodule and re-push this.

File: changelog.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 V.Next
 ----------
+- [MINOR] Added support for http intercept of device code request (#1572)
```
Comment (AdamBJohnsonx) [inline changelog.txt:3]: I'm not sure that this needs a changelog entry, since it's entirely confined to the testing infrastructure.

Comment (fadidurah) [reply changelog.txt:3]: Was running into an error that required an update to the changelog, is there a better way to handle the error?

Comment (AdamBJohnsonx) [reply changelog.txt:3]: Yes.  That is there to make sure you do a gut check on the changes, and include a changelog entry if needed.  If no entry is required, you can add the "No-Changelog" label, and bypass the check.

**Review by AdamBJohnsonx — APPROVED** [review]

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/mocks/MockTokenCreator.java (hunk 1 above)
Comment (shahzaibj) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/mocks/MockTokenCreator.java:67]: Is the tenant always `common`?

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**